From c587fa7c0c14fb08cf5863952da3c0c01b2c7ca5 Mon Sep 17 00:00:00 2001
From: Vladimir Barinov <vladimir.barinov@cogentembedded.com>
Date: Thu, 7 May 2020 18:19:38 +0300
Subject: [PATCH] arm64: dts: r8a77970: isp, csi2, imp switch to generic-uio

This allows to use generic-uio for ISP, CSI2, IMP

Signed-off-by: Vladimir Barinov <vladimir.barinov@cogentembedded.com>
---
 arch/arm64/boot/dts/renesas/r8a77970.dtsi |  74 ++++++++--------
 1 files changed, 53 insertions(+), 52 deletions(-)

diff --git a/arch/arm64/boot/dts/renesas/r8a77970.dtsi b/arch/arm64/boot/dts/renesas/r8a77970.dtsi
index 603efeb..ee44cf7 100644
--- a/arch/arm64/boot/dts/renesas/r8a77970.dtsi
+++ b/arch/arm64/boot/dts/renesas/r8a77970.dtsi
@@ -23,6 +23,8 @@
 		i2c3 = &i2c3;
 		i2c4 = &i2c4;
 		isp0 = &isp0;
+		tisp0 = &tisp0;
+		cisp0 = &cisp0;
 	};
 
 	/* External CAN clock - to be overridden by boards that provide it */
@@ -889,7 +891,7 @@
 		};
 
 		imp_distributer: impdes0 {
-			compatible = "renesas,impx5+-distributer";
+			compatible = "generic-uio";
 			reg = <0 0xffa00000 0 0x10000>;
 			interrupts = <GIC_SPI 281 IRQ_TYPE_LEVEL_HIGH>;
 			clocks = <&cpg CPG_MOD 830>;
@@ -899,97 +901,77 @@
 		};
 
 		imp0 {
-			compatible = "renesas,impx4-legacy";
+			compatible = "generic-uio";
 			reg = <0 0xff900000 0 0x20000>;
-			interrupt-parent = <&imp_distributer>;
-			interrupts = <0>;
 			clocks = <&cpg CPG_MOD 827>;
 			power-domains = <&sysc R8A77970_PD_A2IR0>;
 		};
 
 		imp1 {
-			compatible = "renesas,impx4-legacy";
+			compatible = "generic-uio";
 			reg = <0 0xff920000 0 0x20000>;
-			interrupt-parent = <&imp_distributer>;
-			interrupts = <1>;
 			clocks = <&cpg CPG_MOD 827>;
 			power-domains = <&sysc R8A77970_PD_A2IR0>;
 		};
 
 		imp2 {
-			compatible = "renesas,impx4-legacy";
+			compatible = "generic-uio";
 			reg = <0 0xff940000 0 0x20000>;
-			interrupt-parent = <&imp_distributer>;
-			interrupts = <2>;
 			clocks = <&cpg CPG_MOD 826>;
 			power-domains = <&sysc R8A77970_PD_A2IR1>;
 		};
 
 		imp3 {
-			compatible = "renesas,impx4-legacy";
+			compatible = "generic-uio";
 			reg = <0 0xff960000 0 0x20000>;
-			interrupt-parent = <&imp_distributer>;
-			interrupts = <3>;
 			clocks = <&cpg CPG_MOD 826>;
 			power-domains = <&sysc R8A77970_PD_A2IR1>;
 		};
 
 		impsc0 {
-			compatible = "renesas,impx4-shader";
+			compatible = "generic-uio";
 			reg = <0 0xff980000 0 0x10000>;
-			interrupt-parent = <&imp_distributer>;
-			interrupts = <4>;
 			clocks = <&cpg CPG_MOD 829>;
 			power-domains = <&sysc R8A77970_PD_A2SC0>;
 		};
 
 		impsc1 {
-			compatible = "renesas,impx4-shader";
+			compatible = "generic-uio";
 			reg = <0 0xff990000 0 0x10000>;
-			interrupt-parent = <&imp_distributer>;
-			interrupts = <5>;
 			clocks = <&cpg CPG_MOD 828>;
 			power-domains = <&sysc R8A77970_PD_A2SC1>;
 		};
 
 		impdm0 {
-			compatible = "renesas,impx5-dmac";
+			compatible = "generic-uio";
 			reg = <0 0xffa10000 0 0x1000>;
-			interrupt-parent = <&imp_distributer>;
-			interrupts = <6>;
 			clocks = <&cpg CPG_MOD 825>;
 			power-domains = <&sysc R8A77970_PD_A2DP>;
 		};
 
 		impdm1 {
-			compatible = "renesas,impx5-dmac";
+			compatible = "generic-uio";
 			reg = <0 0xffa11000 0 0x1000>;
-			interrupt-parent = <&imp_distributer>;
-			interrupts = <7>;
 			clocks = <&cpg CPG_MOD 825>;
 			power-domains = <&sysc R8A77970_PD_A2DP>;
 		};
 
 		imppsc0 {
-			compatible = "renesas,impx5+-psc";
+			compatible = "generic-uio";
 			reg = <0 0xffa20000 0 0x4000>;
-			interrupt-parent = <&imp_distributer>;
-			interrupts = <8>;
 			clocks = <&cpg CPG_MOD 825>;
 			power-domains = <&sysc R8A77970_PD_A2DP>;
 		};
 
 		impcnn0 {
-			compatible = "renesas,impx5+-cnn";
+			compatible = "generic-uio";
 			reg = <0 0xff9e0000 0 0x10000>;
-			interrupt-parent = <&imp_distributer>;
-			interrupts = <9>;
 			clocks = <&cpg CPG_MOD 824>;
 			power-domains = <&sysc R8A77970_PD_A2CN>;
 		};
 
 		impc0 {
-			compatible = "renesas,impx4-memory";
+			compatible = "generic-uio";
 			reg = <0 0xed000000 0 0xe0000>;
 			clocks = <&cpg CPG_MOD 830>;
 			power-domains = <&sysc R8A77970_PD_A3IR>;
@@ -1000,7 +982,7 @@
 		};
 
 		imrlx4_ch0: imr0@fe860000 {
-			compatible = "renesas,r8a77970-imr-lx4",
+			compatible = "generic-uio",
 				     "renesas,imr-lx4";
 			reg = <0 0xfe860000 0 0x10000>;
 			interrupts = <GIC_SPI 192 IRQ_TYPE_LEVEL_HIGH>;
@@ -1011,7 +993,7 @@
 		};
 
 		imrlx4_ch1: imr1@fe870000 {
-			compatible = "renesas,r8a77970-imr-lx4",
+			compatible = "generic-uio",
 				     "renesas,imr-lx4";
 			reg = <0 0xfe870000 0 0x10000>;
 			interrupts = <GIC_SPI 193 IRQ_TYPE_LEVEL_HIGH>;
@@ -1022,7 +1004,7 @@
 		};
 
 		imrlx4_ch2: imr2@fe880000 {
-			compatible = "renesas,r8a77970-imr-lx4",
+			compatible = "generic-uio",
 				     "renesas,imr-lx4";
 			reg = <0 0xfe880000 0 0x10000>;
 			interrupts = <GIC_SPI 194 IRQ_TYPE_LEVEL_HIGH>;
@@ -1033,7 +1015,7 @@
 		};
 
 		imrlx4_ch3: imr3@fe890000 {
-			compatible = "renesas,r8a77970-imr-lx4",
+			compatible = "generic-uio",
 				     "renesas,imr-lx4";
 			reg = <0 0xfe890000 0 0x10000>;
 			interrupts = <GIC_SPI 195 IRQ_TYPE_LEVEL_HIGH>;
@@ -1053,6 +1035,22 @@
 			power-domains = <&sysc R8A77970_PD_ALWAYS_ON>;
 		};
 
+		tisp0: tisp0 {
+			compatible = "generic-uio";
+			reg = <0 0xfed00000 0 0x10000>;
+			interrupts = <GIC_SPI 25 IRQ_TYPE_LEVEL_HIGH>;
+			clocks = <&cpg CPG_MOD 817>;
+			power-domains = <&sysc R8A77970_PD_ALWAYS_ON>;
+		};
+
+		cisp0: cisp0 {
+			compatible = "generic-uio";
+			reg = <0 0xfec00000 0 0x20000>;
+			interrupts = <GIC_SPI 26 IRQ_TYPE_LEVEL_HIGH>;
+			clocks = <&cpg CPG_MOD 817>;
+			power-domains = <&sysc R8A77970_PD_ALWAYS_ON>;
+		};
+
 		sadc0: adc@ffce0000 {
 			compatible = "renesas,sadc-r8a77970";
 			reg = <0 0xffce0000 0 0x10000>;
@@ -1209,8 +1207,8 @@
 			resets = <&cpg 603>;
 		};
 
-		csi40: csi2@feaa0000 {
-			compatible = "renesas,r8a77970-csi2";
+		csi40: csi0@feaa0000 {
+			compatible = "renesas,r8a77970-csi2", "generic-uio";
 			reg = <0 0xfeaa0000 0 0x10000>;
 			interrupts = <GIC_SPI 246 IRQ_TYPE_LEVEL_HIGH>;
 			clocks = <&cpg CPG_MOD 716>;
-- 
2.7.4

